R/redcap_dates.R

```diff
@@ -0,0 +1,80 @@
+#' REDCap Date Conversion
+#'
+#' This function is trying to take all (most)
+#' possibilities to enter dates into account and converts them to as.Date
+#' format.
+#'
+#' @param var variable to convert
+#'
+#' @return converted variable
+#' @export
+#' @importFrom crayon bold underline red
+
+#'
+#' @examples
+#' var <- "dataframe$var"
+#' redcap_dates(var)
+
+
+redcap_dates <- function(var) {
+
+  # (as.Date() does not work in ifelse statement -> as.character(as.Date()) does!)
+  # (^ .... $ for exact match)
+  # \\. = 'dot'
+  # between the year format (yy Vs. yyyy) needs to be distinguished (with %y Vs. %Y)
+  # (e.g., 22 with %Y becomes '0022' or 2022 with %y becomes '2020')
+  # for months and days always %d or %m should be used
+  # (e.g., 01.01. with %m = '01.01.' but 1.1. with %D = '01.10.')
+
+  cat(red(bold(underline("Warning:"),
+               "\nThis conversion is only valid with the european date format (days before months) and will lead to wrong results otherwise!!",
+               "\nIf only the year is found, 01.01. is prefixed!",
+               "\nIf only the month and year is found, 01. is prefixed!")))
+
+  # convert to character
+  var <- as.character(var)
+
+  # sometimes entries are separated with / -> replace with .
+  var <- gsub("/",".",var)
```

Comment:
what about dates with hyphens? could happen if it comes from a proper database...

Reply:
done